### PR TITLE
Add configuration model for new Catalog2Registration

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/nuget/nugetgallery/issues/7736.

### CODE REVIEW

File: src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        /// <summary>
+        /// Zero or more URL to the dependency cursor URLs. The registration collector will go no further in the
+        /// catalog than these cursors.
+        /// </summary>
+        public List<string> DependencyCursorUrls { get; set; }
+
+        /// <summary>
+        /// The catalog index URL to poll for package details and package deletes.
+        /// </summary>
+        public string Source { get; set; }
```
Comment (loic-sharma) [inline src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs]: Source may be misunderstood as the service index. What do you think of the name `CatalogIndexUrl`?

File: src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        /// The maximum number of leaf items to allow before pages stop being inlined. In other words, if there are
+        /// less than or equal to this number of package versions for single ID, pages will be inlined in the
+        /// registration index. If there are more than this number of package versions for a single ID, pages will be
+        /// written to their own blobs (i.e. they will no longer be inlined in the index).
+        /// </summary>
+        public int MaxInlinedLeafItems { get; set; } = 127;
+
+        /// <summary>
+        /// Whenever a blob storage write is performed, the code will also ensure that there is at least a single
+        /// snapshot. The snapshot's content is not important but is present to mitigate accidental deletation via
```
Comment (loic-sharma) [inline src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs]: nit: ‘deletion`

File: src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs]: I mimicked the [property name in Catalog2AzureSearch](https://github.com/NuGet/NuGet.Services.Metadata/blob/bc7b6b5f3f84728946785d2021487876b130eb24/src/NuGet.Services.AzureSearch/Catalog2AzureSearch/Catalog2AzureSearchConfiguration.cs#L14) and the other catalog reading jobs (e.g. [Catalog2Dnx](https://github.com/NuGet/NuGet.Services.Metadata/blob/bc7b6b5f3f84728946785d2021487876b130eb24/src/Ng/Jobs/Catalog2DnxJob.cs#L32)). I would rather remain consistent here.

File: src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs]: Done.

**Pull request merged**